python/ray/tune/trial.py

```diff
@@ -275,9 +278,22 @@ def __init__(self,
         self.config = config or {}
         self.local_dir = local_dir  # This remains unexpanded for syncing.
         self.experiment_tag = experiment_tag
-        self.resources = (
-            resources
-            or self._get_trainable_cls().default_resource_request(self.config))
+        trainable_cls = self._get_trainable_cls()
+        if trainable_cls and hasattr(trainable_cls,
+                                     "default_resource_request"):
+            self.resources = trainable_cls.default_resource_request(
+                self.config)
+        else:
+            self.resources = None
+        if self.resources:
+            if resources:
+                raise ValueError(
+                    "Resources for {} have been automatically set to {} "
+                    "by its `default_resource_request()` method. Please "
+                    "clear the `resources_per_trial` option.".format(
+                        trainable_cls, self.resources))
+        else:
+            self.resources = resources or Resources(cpu=1, gpu=0)
```

Comment:
hm can you make this a bit less convoluted?

```
        trainable_cls = self._get_trainable_cls()
        if trainable_cls and hasattr(trainable_cls,
                                     "default_resource_request"):
            default_resources = trainable_cls.default_resource_request(
                self.config)
            if default_resources and resources:
                raise ValueError(
                    "Resources for {} have been automatically set to {} "
                    "by its `default_resource_request()` method. Please "
                    "clear the `resources_per_trial` option.".format(
                        trainable_cls, default_resources))
            resources = default_resources
        self.resources = resources or Resources(cpu=1, gpu=0)
```

Reply:
Done